extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSources.java

```diff
@@ -254,10 +257,11 @@ public AgroalDataSource doCreateDataSource(String dataSourceName) {
             dataSource.setPoolInterceptors(interceptorList);
         }
 
-        if (dataSourceJdbcBuildTimeConfig.telemetry && dataSourceJdbcRuntimeConfig.telemetry.orElse(true)) {
+        if (dataSourceJdbcRuntimeConfig.telemetry.orElse(dataSourceJdbcBuildTimeConfig.telemetry)
```

Comment:
I think using this pattern makes it the intent a bit clearer.

Reply:
@gsmet This condition will pass when OT is enabled at runtime, but disabled at build time. There are steps that need to be done at build time (like creating additional bean you produced)

Reply:
I'm not sure I understand your comment.

Reply:
Thanks for pushing back, I will push an update in a few minutes.

Reply:
Thanks, please consider also moving the DEV mode test as then you can remove the hibernate dependency @gsmet

Reply:
I'm not sure I understand where you want to move the test. Could you take it from there?

Reply:
Sure, will do.